sdk/src/wormhole.ts

```diff
@@ -365,6 +368,48 @@ export class WormholeContext extends MultiProvider<Domain> {
     );
   }
 
+  /**
+   * Gets required fields from a ParsedMessage or ParsedRelayerMessage used to fetch a VAA
+   *
+   * @param txData The ParsedMessage or ParsedRelayerMessage that is the result of a transaction on a source chain
+   * @returns The MessageIdentifier collection of fields to uniquely identify a VAA
+   */
+
+  getMessageIdentifier(
+    txData: ParsedMessage | ParsedRelayerMessage,
+  ): MessageIdentifier {
+    // TODO: wh-connect checks finality first, do we need to?
+    const emitterChain = this.toChainId(txData.fromChain);
+    const emitterAddress = txData.emitterAddress.startsWith('0x')
+      ? txData.emitterAddress.slice(2)
+      : txData.emitterAddress;
+
+    return {
+      emitterChain: emitterChain,
+      emitterAddress,
+      sequence: txData.sequence.toString(),
+    };
+  }
+
+  /**
+   * Gets a VAA from the API or Guardian RPC
+   *
+   * @param msg The MessageIdentifier used to fetch the VAA
+   * @returns The ParsedVAA if available
+   */
+  async getVAA(msg: MessageIdentifier): Promise<ParsedVaa | undefined> {
+    const { emitterChain, emitterAddress, sequence } = msg;
+    const url = `${this.conf.api}/api/v1/vaas/${emitterChain}/${emitterAddress}/${sequence}`;
+    const response = await axios.get(url);
+
+    // TODO: raise exception? wait?
+    if (!response.data.data) return;
```

Comment:
You could try using `getSignedVaaWithRetry` ([source code](https://github.com/wormhole-foundation/wormhole/blob/8e0cf4c31f39b5ba06b0f6cdb6e690d3adf3d6a3/sdk/js/src/rpc/getSignedVAAWithRetry.ts)) to query the guardian rpc as a fallback

Reply:
should I? What is the expected behavior for this fn? that it waits until its available or returns immediately if not found?